### DISCUSSION TITLE
Cannot access private repos with API with Personal Access Token

### DISCUSSION THREAD

**Discussion by force2b** [community/community#30146] [Apps, API and Webhooks] [answered]
We have a few private repos in our org. When I call the api to retrieve the release information, I get a not-found response:

Command: `curl -H 'Authorization: {authtoken} OAUTH-TOKEN' -s "https://api.github.com/repos/MyOrg/Transport-Layer/releases/latest"`

Response:
```
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-the-latest-release"
}
```

I've tested my token against all the public repos. They work fine.

The perms for the auth token are: ead:discussion, read:enterprise, read:org, read:packages, read:project, read:public_key, read:repo_hook, read:user, repo. And I used the [Configure SSO] button to authorize access to our organization.

Is there something else missing to allow the api call (via Curl) to retrieve release info on the repository using only a Personal Access Token? Or do I have to go down the full oAuth route (which I'm hoping to avoid)?

**Comment by byrneh** [comment, marked as answer]
Are your sure you access token is being provided correctly in the cURL command to the GitHub Server. If its is not that woud explain why you can see public repos (does not require authentication) and not your private repos.

A good check to confirm is to amend the same curl command to access the user endpoint (the user the token belongs to). 
`https://api.github.com/user`
if you get a respose `"message": "Requires authentication"` you know there is a problem with the token value or how you have provided it to your cURL command.

**Comment by force2b** [comment]
Thanks for the response. Looks like the OAUTH-TOKEN syntax was not correct, or I had the header content malformed. The following works perfectly:
`curl -u force2b:{token} -s https://api.github.com/repos/MyOrg/Transport-Layer/releases/latest`
